core/src/main/java/org/infinispan/marshall/persistence/impl/MarshalledEntryFactoryImpl.java

```diff
@@ -63,7 +63,11 @@ public MarshallableEntry create(Object key, Object value) {
    @Override
    public MarshallableEntry create(Object key, Object value, Metadata metadata, PrivateMetadata internalMetadata,
          long created, long lastUsed) {
-      return new MarshallableEntryImpl<>(key, value, metadata, internalMetadata, created, lastUsed, marshaller);
+      PrivateMetadata privateMetadataToUse = internalMetadata != null && !internalMetadata.isEmpty() ? internalMetadata : null;
```

Comment:
I would reverse the condition since only xsite and transactions have `internalMetadata`
```suggestion
      PrivateMetadata privateMetadataToUse = internalMetadata == null  !! internalMetadata.isEmpty() ? null : internalMetadata;
```

Reply:
It is the same logic wise. I feel the way I have it is easier to read personally.